dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -1035,6 +1042,85 @@ export function JsonRpcContextProvider({
     ),
   };
 
+   // -------- TRON RPC METHODS --------
+
+  const tronRpc = {
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // Nile TestNet, if you want to use in MainNet, change the fullHost to 'https://api.trongrid.io'
+        const tronWeb = new TronWeb({
+          fullHost: 'https://nile.trongrid.io/',
+        })
+
+        const testTransaction = await tronWeb.transactionBuilder.triggerSmartContract(
+          'TXLAQ63Xg1NAzckPwKHvzw7CSEmLMEqcdj',
+          'approve(address,uint256)',
+          { feeLimit: 200000000 },
+          [
+            { type: 'address', value: address },
+            { type: 'uint256', value: 0 }
+          ],
+          address
+        );
+
+        try {
+          const { result } = await client!.request<{ result: any }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_TRON_METHODS.TRON_SIGN_TRANSACTION,
+              params: {
+                address,
+                transaction:{
+                  ...testTransaction
+                }
+              }
+            }
+          });
+          
+          return {
+            method: DEFAULT_TRON_METHODS.TRON_SIGN_TRANSACTION,
+            address,
+            valid: true,
+            result: result.signature
+          };
+        } catch (error: any) {
+          throw new Error(error);
+        }
+      }
+    ),
+    testSignMessage: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+
+        var message = 'TestTronString';
```

Comment:
Code style nit: please use `let` over `var`. Looks like message is never reassigned so suggesting `const`:
```suggestion
        const message = 'This is a message to be signed for Tron';
```

Reply:
change the code as above